### PR TITLE
bpo-36797: Prune more legacy distutils documentation

### PR DESCRIPTION
**Pull request by ncoghlan**
https://bugs.python.org/issue36797

### CODE REVIEW

**Comment by ncoghlan** [comment]
@pganssle Reviewing @kojoidrissa's change in #13087 prompted me to make the markings on the rest of the legacy distutils docs a bit more emphatic in saying "This probably isn't the page you want to be looking at"

**Comment by pganssle** [comment]
Hm, would it be worth it to move this note into a separate stub and use `.. include::` instead of copy-pasting it everywhere?

Given that there's a URL in it, it might be worth it to have a canonical stub for this.

**Comment by ncoghlan** [comment]
@pganssle Yeah, that would make a lot of sense.

**Review by eamanu — APPROVED** [review]
LGTM

File: Doc/distutils/apiref.rst [hunk 1]
```diff
@@ -4,6 +4,16 @@
 API Reference
 *************
 
+.. seealso::
+
+   `New and changed setup.py arguments in setuptools <setuptools-setup-py>`_
```
Comment (hroncok) [inline Doc/distutils/apiref.rst:9]: @ncoghlan This is a dead link to https://docs.python.org/3.8/distutils/setuptools-setup-py

See linkchecker output:

```

URL        `setuptools-setup-py'
Name       `New and changed setup.py arguments in setuptools'
Parent URL file:///builddir/build/BUILD/Python-3.8.0b4/Doc/build/html/distutils/apiref.html, line 106, col 5
Real URL   file:///builddir/build/BUILD/Python-3.8.0b4/Doc/build/html/distutils/setuptools-setup-py
Check time 0.001 seconds
Result     Error: URLError: <urlopen error [Errno 2] No such file or directory: '/builddir/build/BUILD/Python-3.8.0b4/Doc/build/html/distutils/setuptools-setup-py'>
 4 threads active,     0 links queued, 4445 links in 4449 URLs checked, runtime 1 minute, 46 seconds
```

Where is this actually supposed to lead to?

File: Doc/distutils/apiref.rst [hunk 2]
```diff
@@ -4,6 +4,16 @@
...
+      The ``setuptools`` project adds new capabilities to the ``setup`` function
+      and other APIs, makes the API consistent across different Python versions,
+      and is hence recommended over using ``distutils`` directly.
+
+.. _setuptools-setup-py: https://setuptools.readthedocs.io/en/latest/setuptools.html#new-and-changed-setup-keywords
```
Comment (hroncok) [inline Doc/distutils/apiref.rst:14]: Oh I see, this should be the link. I'll prep a PR.

Comment (hroncok) [reply Doc/distutils/apiref.rst:14]: https://github.com/python/cpython/pull/15700

**Pull request merged**